### DISCUSSION TITLE
develop/utils/video/

### DISCUSSION THREAD

**Discussion by giscus** [roboflow/supervision#1607] [Docs]
# develop/utils/video/

A set of easy-to-use utilities that will come in handy in any computer vision project.

https://supervision.roboflow.com/develop/utils/video/

**Comment by Muhammad-237** [comment]
How make inference on livestream or cctv footage with ip address can anyone help me and write some code snippet that help me with that

**Reply by LinasKo** [reply]
Hi @Muhammad-237 👋 

I don't know the specifics of your stream. Our `inference` supports RTSP streams, accepting a stream URL in [`InferencePipeline`](https://inference.roboflow.com/using_inference/inference_pipeline/).

Here are a few other articles about it:
* https://roboflow.com/use-model-with-rtsp/yolov8
* https://inference.roboflow.com/quickstart/run_model_on_rtsp_webcam/

Note that we typically cannot help debug issues related to connecting with your stream.

**Comment by brockt96** [comment]
I think that the FPSMonitor's documentation needs a bit of updating. In the example code, it says that fps = fps_monitor(). But my understanding is that fps_monitor is not callable. Instead, you get the fps by: fps = fps_monitor.fps. Just thought I'd let y'all know.

**Reply by LinasKo** [reply]
Hi @brockt96 👋 

Thanks for letting us know! Which version of the docs are you on? I did a quick scan through [0.24.0](https://supervision.roboflow.com/latest/utils/video/), and couldn't see any `fps_monitor()` calls. Still, I might be overlooking something 😉

**Reply by brockt96** [reply]
Not sure, but it was likely an older one. Thanks for posting the link!

**Reply by LinasKo** [reply]
No worries. We'd rather have a few mistaken reports than incorrect docs

**Comment by AbelAbeb** [comment]
hello guys, I have one question. regardless of the fps of the video how can we adjust/ fix fps to constant number and generate the frames based on that. let say I need 15 fps every 4 microseconds, how can I adjust that.
